src/Framework/AspNetCoreAnalyzers/src/Analyzers/Infrastructure/WellKnownTypes.cs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.App.Analyzers.Infrastructure;
+
+internal enum WellKnownType
+{
+    Microsoft_AspNetCore_Components_Rendering_RenderTreeBuilder,
+    Microsoft_AspNetCore_Http_Metadata_IFromBodyMetadata,
+    Microsoft_AspNetCore_Http_Metadata_IFromFormMetadata,
+    Microsoft_AspNetCore_Http_Metadata_IFromHeaderMetadata,
+    Microsoft_AspNetCore_Http_Metadata_IFromQueryMetadata,
+    Microsoft_AspNetCore_Http_Metadata_IFromServiceMetadata,
+    Microsoft_AspNetCore_Routing_IEndpointRouteBuilder,
+    Microsoft_AspNetCore_Mvc_ControllerAttribute,
+    Microsoft_AspNetCore_Mvc_NonControllerAttribute,
+    Microsoft_AspNetCore_Mvc_NonActionAttribute,
+    Microsoft_AspNetCore_Http_AsParametersAttribute,
+    System_Threading_CancellationToken,
+    Microsoft_AspNetCore_Http_HttpContext,
+    Microsoft_AspNetCore_Http_HttpRequest,
+    Microsoft_AspNetCore_Http_HttpResponse,
+    System_Security_Claims_ClaimsPrincipal,
+    Microsoft_AspNetCore_Http_IFormFileCollection,
+    Microsoft_AspNetCore_Http_IFormFile,
+    System_IO_Stream,
+    System_IO_Pipelines_PipeReader,
+    System_IFormatProvider,
+    System_Uri,
+    Microsoft_AspNetCore_Builder_ConfigureHostBuilder,
+    Microsoft_AspNetCore_Builder_ConfigureWebHostBuilder,
+    Microsoft_Extensions_Hosting_GenericHostWebHostBuilderExtensions,
+    Microsoft_AspNetCore_Hosting_WebHostBuilderExtensions,
+    Microsoft_AspNetCore_Hosting_HostingAbstractionsWebHostBuilderExtensions,
+    Microsoft_Extensions_Hosting_HostingHostBuilderExtensions,
+    Microsoft_AspNetCore_Builder_EndpointRoutingApplicationBuilderExtensions,
+    Microsoft_AspNetCore_Builder_WebApplication,
+    Microsoft_AspNetCore_Builder_EndpointRouteBuilderExtensions,
+    System_Delegate,
+    Microsoft_AspNetCore_Mvc_ModelBinding_IBinderTypeProviderMetadata,
+    Microsoft_AspNetCore_Mvc_BindAttribute,
+    Microsoft_AspNetCore_Http_IResult,
+    Microsoft_AspNetCore_Mvc_IActionResult,
+    Microsoft_AspNetCore_Mvc_Infrastructure_IConvertToActionResult,
+    Microsoft_AspNetCore_Http_RequestDelegate,
+    System_Threading_Tasks_Task_T,
+}
+
+internal class WellKnownTypes
+{
+    private static readonly ConditionalWeakTable<Compilation, WellKnownTypes> _compilationToLazyWellKnownTypes = new();
```

Comment:
⚠️ We've had trouble with CWTs with keys holding large object graphs. I recommend modeling after this:
https://github.com/dotnet/roslyn-analyzers/blob/main/src/Utilities/Compiler/WellKnownTypeProvider.cs

Reply:
Done